test_end2end/collapse_test/test_direct_nodes_same_level/test.yml

```diff
@@ -0,0 +1,47 @@
+query: >
+  MATCH (n:P) WITH COLLECT(n) AS nodes
+  CALL collapse.collapse(nodes, ['Edge'])
+  YIELD from_vertex, path, to_vertex
+  RETURN from_vertex, nodes(path) AS path, to_vertex;
+
+output:
+  - from_vertex:
+      labels:
+      - P
+      properties:
+        id: p1
+    path:
+      - labels:
+        - P
+        properties:
+          id: p1
+      - labels:
+        - P
+        properties:
+          id: p2
+    to_vertex:
+      labels:
+      - P
+      properties:
+        id: p2
+
+  - from_vertex:
+      labels:
+      - P
+      properties:
+        id: p2
+    path:
+      - labels:
+        - S
+        properties:
+          id: s2
+      - labels:
+        - S
+        properties:
+          id: s1
+    to_vertex:
+      labels:
+      - P
+      properties:
+        id: p1
```

Comment:
This looks a bit weird because the whole format has a lot of repetitions. Consider some more tabular format, e.g., header list + an array for each Bolt record?

Reply:
```output
# Proposition #1
- from_vertex:
    labels: [P]
    properties: {id: p2}
  path:
    - labels: [S]
      properties: {id: s3}
    - labels: [S]
      properties: {id: s2}
  to_vertex:
    labels: [P]
    properties: {id: p1}
    
# Proposition #2
- from_vertex:
    _labels: [P]
    id: p2
  nodes(path):
  - _labels: [S]
    id: s3
  - _labels: [S]
    id: s2
  to_vertex:
    _labels: [P]
    id: p1

Reply:
By default, YAML is able to minimize leaf nodes to the block style. Still, there are some repetitions, any suggestion on this is welcome.

Reply:
Proposition #1 seems the best one out of the proposed ones :smile: